File: Question-Three/transport.js
let routes = ["Route A", "Route B"];
let passengerVolume = { peak: 120, offPeak: 40 };
let specialEvent = true;
let maintenanceTime = "02:00 - 04:00";
let demandPattern = { "Route A": 80, "Route B": 100 };
function optimizeSchedule() {
    if (passengerVolume.peak > 100) {
        console.log("Increase service during peak hours");
    }
    if (passengerVolume.offPeak < 50) {
        console.log("Reduce service during off-peak hours");
    }
    if (specialEvent) {
        console.log("Add extra trips and modify routes for the event");
    }
    console.log("Schedule maintenance at: " + maintenanceTime);
    console.log("Suggested route: Route B (higher demand)");
}
optimizeSchedule();

File: Question-Three/transport.kt
val routes = listOf("Route A", "Route B")
val passengerVolume = mapOf("peak" to 120, "off_peak" to 40)
val specialEvent = true
val maintenanceTime = "02:00 - 04:00"
val demandPattern = mapOf("Route A" to 80, "Route B" to 100)
fun optimizeSchedule() {
    if (passengerVolume["peak"]!! > 100) {
        println("Increase service during peak hours")
    }
    if (passengerVolume["off_peak"]!! < 50) {
        println("Reduce service during off-peak hours")
    }
    if (specialEvent) {
        println("Add extra trips and modify routes for the event")
    }
    println("Schedule maintenance at: $maintenanceTime")
    println("Suggested route: Route B (higher demand)")
}
fun main() {
    optimizeSchedule()
}